packages/media-download/src/yt-dlp.ts

```diff
@@ -0,0 +1,35 @@
+import fs from 'node:fs';
+import { $ } from 'zx';
+
+export type MediaMetadata = {
+	title: string;
+	extension: string;
+	filename: string;
+	mediaPath: string;
+};
+
+const extractInfoJson = (infoJsonPath: string): MediaMetadata => {
+	const file = fs.readFileSync(infoJsonPath, 'utf8');
+	const json = JSON.parse(file);
+	return {
+		title: json.title,
+		extension: json.ext,
+		filename: json.filename,
+		mediaPath: `${json.filename}.${json.ext}`,
+	};
+};
+
+export const downloadMedia = async (
+	url: string,
+	destinationDirectoryPath: string,
+	id: string,
+) => {
+	const output =
+		await $`yt-dlp --write-info-json --no-clean-info-json --newline -o "${destinationDirectoryPath}/${id}" ${url}`;
```

Comment:
has yt-dlp been pretty consistent for youtube? I was noticing some flakiness using yt-dlp to download videos today.

Reply:
as discussed, I've had some success with it. I'm hopeful that even if we end up having to use a different tool most of this code will still be useful